### PR TITLE
ARTEMIS-2990 - update wildcard address map on creation only, avoid du…

### PR DESCRIPTION
**Pull request by gtully**
…plicates and duplicate checks

### CODE REVIEW

File: artemis-server/src/main/java/org/apache/activemq/artemis/core/postoffice/impl/AddressImpl.java [hunk 1]
```diff
@@ -69,24 +68,16 @@ public boolean containsWildCard() {
...
 
    @Override
    public void addLinkedAddress(final Address address) {
-      if (linkedAddresses == null) {
-         linkedAddresses = new ArrayList<>(1);
-      }
-      if (!linkedAddresses.contains(address)) {
-         linkedAddresses.add(address);
-      }
+      linkedAddresses.add(address);
```
Comment (franz1981) [inline artemis-server/src/main/java/org/apache/activemq/artemis/core/postoffice/impl/AddressImpl.java:76]: I would add an `assert !linkedAddress,contains(address)` here as an invariant on the tests

**Comment by michaelandrepearce** [comment]
@gtully polite ask. Can you please let prs open for at least a day or two for people to have time to review before merging. This pr was raised and merged in only a few ours not allowing anyone time to review

File: artemis-server/src/main/java/org/apache/activemq/artemis/core/postoffice/impl/AddressImpl.java [hunk 2]
```diff
@@ -37,7 +36,7 @@
 
    private final boolean containsWildCard;
 
-   private List<Address> linkedAddresses = null;
+   private final List<Address> linkedAddresses = new ArrayList<>();
```
Comment (michaelandrepearce) [inline artemis-server/src/main/java/org/apache/activemq/artemis/core/postoffice/impl/AddressImpl.java:39]: Whats the impact here for those not using wildcards

Comment (gtully) [reply artemis-server/src/main/java/org/apache/activemq/artemis/core/postoffice/impl/AddressImpl.java:39]: there is an impact, it is reverted to lazy init via https://github.com/apache/activemq-artemis/pull/3343 where a set works better for deletions.

**Comment by gtully** [comment]
@michaelandrepearce polite ack.

File: artemis-server/src/main/java/org/apache/activemq/artemis/core/postoffice/impl/AddressImpl.java (hunk 1 above)
Comment (michaelandrepearce) [reply artemis-server/src/main/java/org/apache/activemq/artemis/core/postoffice/impl/AddressImpl.java:76]: Dont add assert into runtime code.

Comment (michaelandrepearce) [reply artemis-server/src/main/java/org/apache/activemq/artemis/core/postoffice/impl/AddressImpl.java:76]: It is better to follow the pattern from set interface and return true / false if operation actually actioned

**Pull request merged**